src/engine/enginemaster.cpp

```diff
@@ -726,67 +760,77 @@ void EngineMaster::process(const int iBufferSize) {
     }
 
     if (masterEnabled) {
-        m_pMasterDelay->process(m_pMaster, m_iBufferSize);
+        m_pMasterDelay->process(m_pMaster, iBufferSize);
     } else {
-        SampleUtil::clear(m_pMaster, m_iBufferSize);
+        SampleUtil::clear(m_pMaster, iBufferSize);
     }
     if (headphoneEnabled) {
-        m_pHeadDelay->process(m_pHead, m_iBufferSize);
+        m_pHeadDelay->process(m_pHead, iBufferSize);
     }
     if (boothEnabled) {
-        m_pBoothDelay->process(m_pBooth, m_iBufferSize);
+        m_pBoothDelay->process(m_pBooth, iBufferSize);
     }
 
     // We're close to the end of the callback. Wake up the engine worker
     // scheduler so that it runs the workers.
     m_pWorkerScheduler->runWorkers();
 }
 
-void EngineMaster::applyMasterEffects() {
+void EngineMaster::applyMasterEffects(int iBufferSize) {
     // Apply master effects
     if (m_pEngineEffectsManager) {
         GroupFeatureState masterFeatures;
         masterFeatures.has_gain = true;
         masterFeatures.gain = m_pMasterGain->get();
-        m_pEngineEffectsManager->processPostFaderInPlace(m_masterHandle.handle(),
-                                                         m_masterHandle.handle(),
-                                                         m_pMaster,
-                                                         m_iBufferSize, m_iSampleRate,
-                                                         masterFeatures);
+        m_pEngineEffectsManager->processPostFaderInPlace(
+                m_masterHandle.handle(),
+                m_masterHandle.handle(),
+                m_pMaster,
+                iBufferSize,
+                m_iSampleRate,
+                masterFeatures);
     }
 }
 
-void EngineMaster::processHeadphones(const CSAMPLE_GAIN masterMixGainInHeadphones) {
+void EngineMaster::processHeadphones(
+        const CSAMPLE_GAIN masterMixGainInHeadphones,
+        int iBufferSize) {
     // Add master mix to headphones
-    SampleUtil::addWithRampingGain(m_pHead, m_pMaster,
-                                   m_headphoneMasterGainOld,
-                                   masterMixGainInHeadphones, m_iBufferSize);
+    SampleUtil::addWithRampingGain(
+            m_pHead,
+            m_pMaster,
+            m_headphoneMasterGainOld,
+            masterMixGainInHeadphones,
+            iBufferSize);
     m_headphoneMasterGainOld = masterMixGainInHeadphones;
 
     // If Head Split is enabled, replace the left channel of the pfl buffer
     // with a mono mix of the headphone buffer, and the right channel of the pfl
     // buffer with a mono mix of the master output buffer.
     if (m_pHeadSplitEnabled->toBool()) {
         // note: NOT VECTORIZED because of in place copy
-        for (unsigned int i = 0; i + 1 < m_iBufferSize; i += 2) {
+        for (SINT i = 0; i + 1 < iBufferSize; i += 2) {
             m_pHead[i] = (m_pHead[i] + m_pHead[i + 1]) / 2;
             m_pHead[i + 1] = (m_pMaster[i] + m_pMaster[i + 1]) / 2;
         }
```

Comment:
I was curious so I double checked the comment and its not valid actually (neither gcc, clang or even notoriously lazy msvc). https://compiler-explorer.com/z/z5xv7Ye3b

Reply:
This loop is not vectorized with: 
* clang 13 https://compiler-explorer.com/z/jfKMPEbh1 
* gcc 12.2 https://compiler-explorer.com/z/rG65r6GTM
* mcvc 19 https://compiler-explorer.com/z/6bPY1fMxx
But with to some extend 
* clang 14 https://compiler-explorer.com/z/16nfzaK3W

I will update the comment.

Reply:
Nevermind, I was judging by eye from the generated assembly, ignore my comment.